### ISSUE TITLE
match-sorter-utils const characterMap causes JavaScript syntax error

### ISSUE THREAD

**Issue by CoryWalker1** [TanStack/table#4839]
### Describe the bug

After bundling my app using Rollup, both FF and Edge report that a syntax error is occurring at this line:

```
/**
	 * match-sorter-utils
	 *
	 * Copyright (c) TanStack
	 *
	 * This source code is licensed under the MIT license found in the
	 * LICENSE.md file in the root directory of this source tree.
	 *
	 * @license MIT
	 */
	const characterMap = {
	  Ã€: 'A', // ILLEGAL CHARACTER HERE.  IT PROBABLY DOESN'T LIKE Ã€.
	  Ã: 'A',
```

The illegal character being reported is Ã€.  I believe this characterMap is written in TypeScript and defined remove-accents.ts.

```
const characterMap: Record<string, string> = {
  À: 'A',
```

### Your minimal, reproducible example

https://github.com/TanStack/table/blob/main/packages/match-sorter-utils/src/remove-accents.ts

### Steps to reproduce

All I'm doing is using the Material React Table in the Basic Example.

```
import MaterialReactTable from 'material-react-table';

<MaterialReactTable columns={columns} data={data} />
```

### Expected behavior

The expectation is that these characters would not be reported as illegaly and that the bundle parses in the browser.

### How often does this bug happen?

Every time

### Screenshots or Videos

<img width="464" alt="IllegalCharacter" src="https://user-images.githubusercontent.com/130489168/236543950-ea4382c8-0c96-4373-8391-5af91489e566.png">
<img width="956" alt="SyntaxError" src="https://user-images.githubusercontent.com/130489168/236544119-dd97aae5-99b3-47bc-a7c9-01785b81175a.png">

### Platform

Windows 11, Edge, FireFox

### react-table version

v1.11.2

### Additional context

```
{
  "dependencies": {
    "@emotion/react": "^11.10.8",
    "@emotion/styled": "^11.10.8",
    "@mui/icons-material": "^5.11.16",
    "@mui/material": "^5.12.3",
    "@tanstack/match-sorter-utils": "^8.8.4",
    "axios": "^1.4.0",
    "material-react-table": "^1.11.2",
    "q": "^1.5.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  }
}

```

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by CoryWalker1** [comment]
It appears that this characterMap variable passes parsing if the field name is in quotes.

```
const characterMap = {
	  'Ã€': 'A',  // THIS WORKS AND DOES NOT CAUSE A PARSING ERROR IN THE BROWSER.
```

**Comment by CoryWalker1** [comment]
Hello, it's been 5 days.  Is this the place to report issues and potential bugs?

**Comment by KevinVandy** [comment]
Yes, but I think this bug only happens if you import directly from src. No one else I know had had the same problem

**Comment by riotrah** [comment]
This issue causes our clojurescript app to break when trying to use tanstack query dev tools. 

It's out of our "userland" hands as to how `node_modules `are resolved (if it's relevant, we use `shadow-cljs` to bundle/build our app). 

If the fix is only to add quote marks (I'm uncertain if it is for our particular case, could probably test this and report back), I really hope this can be done or that a PR I might open would not be rejected outright. 

Let me know if I should create a new issue. Thanks

**Comment by riotrah** [comment]
Confirming that making these keys quoted fixed it
